Transfer to the new repo

LightnessOfBeing opened the pull request.

LightnessOfBeing committed:
Transfer to the new repo

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/Main.java:
```diff
@@ -0,0 +1,21 @@
+
+public class Main {
+
+    public static void main(String[] args){
```
Перед "{" нужен пробел, здесь и ниже

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/Main.java:
```diff
@@ -0,0 +1,21 @@
+        m.sortMatrixByColumns();
+        m.simpleOutput();
+
+    }
```
А перед "}" пустая строчка обычно не ставится

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/SpiralMatrix.java:
```diff
@@ -0,0 +1,185 @@
+     * n - denotes size of array.
+     */
+
+    private final int n;
```
Размер массива можно не хранить, массив сам его знает

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/SpiralMatrix.java:
```diff
@@ -0,0 +1,185 @@
+     * @param n - size of matrix.
+     */
+
+    SpiralMatrix(int n, int[][] array){
```
Класс паблик, а конструктор имеет пакетную видимость, так и задумывалось?

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/SpiralMatrix.java:
```diff
@@ -0,0 +1,185 @@
+        matrix = new int[n][n];
+
+        for (int i = 0; i < n; i++) {
+            System.arraycopy(array[i], 0, matrix[i], 0, n);
```
Надо бы ещё проверять, что нам передали, и если что, адекватно ругаться

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/SpiralMatrix.java:
```diff
@@ -0,0 +1,185 @@
+     * I wanted to make this method private, but in this case there is no option to test this method.
+     */
+
+    void sortMatrixByRows(){
```
Эта штука тоже, наверное, должна быть public, она упоминается в условии задачи

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/SpiralMatrix.java:
```diff
@@ -0,0 +1,185 @@
+     * I wanted to make this method private, but in this case there is no option to test this method.
+     */
+
+    void transposeMatrix(){
```
Пакетная видимость по какой-то причине, или просто видимость не указана?

yurii-litvinov commented on gradleSpiralMatrix/src/main/java/ru/spbau/mit/vishnyakov/SpiralMatrix.java:
```diff
@@ -0,0 +1,185 @@
+
+    }
+
+    public int[][] getMatrix(){
```
О, паблик-метод без комментария :)

LightnessOfBeing replied:
Нам на одной из смерженных практик Тимофей говорил, что у геттеров и сеттеров можно не писать комментарии.

yurii-litvinov commented on gradleSpiralMatrix/src/test/java/ru/spbau/mit/vishnyakov/SpiralMatrixTest.java:
```diff
@@ -0,0 +1,154 @@
+        assertArrayEquals(new int[]{5, 5, 5}, m.getMatrix()[2]);
+
+    }
+
```
Ещё надо бы плохие случаи проверить, когда передаётся заведомо неправильная матрица (например, `new int[][]{{1, 2, 3}, {4, 5}, {6}}` или null), программа должна адекватно реагировать на такие ситуации

yurii-litvinov reviewed (changes requested):
Бинарники в гит не надо выкладывать. Могу порекомендовать вот это: https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore плюс вот это: https://github.com/github/gitignore/blob/master/Gradle.gitignore плюс вот это: https://github.com/github/gitignore/blob/master/Java.gitignore

LightnessOfBeing committed:
Updated Spiral Matrix.

Added code which handle incorrect input, also added tests which handles such cases.
Fixed all style complaints.

yurii-litvinov commented on gradleSpiralMatrix/build/reports/tests/test/classes/ru.spbau.mit.vishnyakov.SpiralMatrixTest.html:
```diff
@@ -0,0 +1,151 @@
+<!DOCTYPE html>
```
Это тоже надо удалить, это что-то сгенерённое

yurii-litvinov commented on gradleSpiralMatrix/build/reports/tests/test/css/base-style.css:
```diff
@@ -0,0 +1,179 @@
+
```
И это

yurii-litvinov reviewed (changes requested):
Надо ещё лишние файлы поудалять, даже без .gitignore, вручную просто. В остальном теперь всё ок

LightnessOfBeing committed:
Deleted auto-generated files.

Deleted unnecessary auto-generated files.

yurii-litvinov reviewed (changes requested):
Не-а, надо вообще всю папку build грохнуть

LightnessOfBeing committed:
Deleted build folder.

yurii-litvinov reviewed (approved):
Ага, теперь всё так, зачтена
